### PR TITLE
Interactive Viewer Features and Lighting

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

To debug and experiment with Habitat-sim interactive features it is useful to have a few new features. This PR cleans up some unused or irrelevant Viewer functions/key bindings and adds:
- a mouse mode toggled by `'m'` to switch between looking and grabbing UI. Looking UI is default Viewer behavior while grabbing UI creates constraints between the mouse pointer and a clicked object. Left click creates a point-to-point constraint and right click creates a rigid constraint.
- instance a URDF from an specified path input with `'t'`
- override the default light setup from a specific LightingAttributes config local to the viewer `src/utils/viewer/default_light_override.lighting_config.json` with the `'L'` key for quick lighting iteration. 

Secondary but related to viewer changes is a modified default light setup with directional lights from all directions.

## How Has This Been Tested

Updated render testing images.
Local testing of viewer. Example in ReplicaCAD:

https://user-images.githubusercontent.com/1445143/123737853-939a0b80-d858-11eb-8f56-c857acaf2a60.mov

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by jturner65 — APPROVED** [review]
LGTM! Very nice!

File: src/esp/physics/ArticulatedObject.h [hunk 1]
```diff
@@ -783,7 +792,7 @@ class ArticulatedObject : public esp::physics::PhysicsObjectBase {
   //=========== END - Joint Motor API ===========
 
   //! map PhysicsManager objectId to local multibody linkId
-  std::map<int, int> objectIdToLinkId_;
+  std::unordered_map<int, int> objectIdToLinkId_;
```
Comment (Skylion007) [inline src/esp/physics/ArticulatedObject.h:795]: Nice! Keep up the conversions. :)

File: src/utils/viewer/viewer.cpp [hunk 2]
```diff
@@ -88,6 +88,50 @@ std::string getCurrentTimeString() {
...
+  int constraintId = esp::ID_UNDEFINED;
+  esp::sim::Simulator& sim_;
+
+  float gripDepth = 0;
+
+  MouseGrabber(const esp::physics::RigidConstraintSettings& settings,
+               float _gripDepth,
+               esp::sim::Simulator& sim)
+      : sim_(sim), settings_(settings) {
+    gripDepth = _gripDepth;
```
Comment (Skylion007) [inline src/utils/viewer/viewer.cpp:114]: Should be iin member nitializer list, no?

File: src/utils/viewer/viewer.cpp [hunk 3]
```diff
@@ -88,6 +88,50 @@ std::string getCurrentTimeString() {
...
+    constraintId = sim_.createRigidConstraint(settings_);
```
Comment (Skylion007) [inline src/utils/viewer/viewer.cpp:115]: Likewise

**Review by Skylion007 — APPROVED** [review]

**Pull request merged**